### PR TITLE
LPD-24309 Support relationships in FormInputMappingOptions

### PR DESCRIPTION
**Pull request by sandrodw3**

### CODE REVIEW

File: modules/apps/layout/layout-content-page-editor-web/src/main/resources/META-INF/resources/page_editor/common/components/MappingSelector.js [hunk 1]
```diff
@@ -273,6 +276,47 @@ function MappingSelector({
...
+				label: Liferay.Language.get('relationship'),
+				value: MAPPING_SOURCE_TYPES.relationship,
+			});
+		}
+
+		types.push({
+			label: Liferay.Language.get('specific-content'),
+			value: MAPPING_SOURCE_TYPES.content,
+		});
+
```
Comment (victorg1991) [inline modules/apps/layout/layout-content-page-editor-web/src/main/resources/META-INF/resources/page_editor/common/components/MappingSelector.js:345]:
```javascript
const types = [
		{
			label: sub(
				Liferay.Language.get('x-default'),
				selectedMappingTypes.subtype
					? selectedMappingTypes.subtype.label
					: selectedMappingTypes.type.label
			),
			value: MAPPING_SOURCE_TYPES.structure,
		},

		...(relationships?.length && Liferay.FeatureFlags['LPD-20213']
			? [
					{
						label: Liferay.Language.get('relationship'),
						value: MAPPING_SOURCE_TYPES.relationship,
					},
			  ]
			: []),
		{
			label: Liferay.Language.get('specific-content'),
			value: MAPPING_SOURCE_TYPES.content,
		},
	];


😏
```

File: modules/apps/layout/layout-content-page-editor-web/src/main/resources/META-INF/resources/page_editor/app/utils/editable_value/isMappedToRelationship.ts [hunk 2]
```diff
@@ -0,0 +1,24 @@
...
+ */
+
+type Editable = {
+	classNameId?: string;
+	mappedField?: string;
+};
+
+type MappedToRelationship = {
+	classNameId?: string;
+	mappedField?: string;
```
Comment (victorg1991) [inline modules/apps/layout/layout-content-page-editor-web/src/main/resources/META-INF/resources/page_editor/app/utils/editable_value/isMappedToRelationship.ts:13]: these shouldn't be optional right? if we are checking its existence

Comment (sandrodw3) [reply modules/apps/layout/layout-content-page-editor-web/src/main/resources/META-INF/resources/page_editor/app/utils/editable_value/isMappedToRelationship.ts:13]: Yeeees, my bad 😇

**Comment by sandrodw3** [comment]
ci:forward

**Comment by ealonso** [comment]
ci:forward:force

**Pull request closed without merging**